tests/builders/metrics_regex_with_linenum.yml

```diff
@@ -0,0 +1,29 @@
+buildspecs:
+  metric_regex_example:
+    executor: generic.local.bash
+    type: script
+    description: capture result metric from output
+    run: |
+      echo "HPCG result is VALID with a GFLOP/s rating of=63.6515"
+      echo "HPCG result is SUPERB with a GFLOP/s rating of=88"
+    tags: tutorials
+    metrics:
+      hpcg_rating_stream:
+        type: float
+        regex:
+          exp: '(\d+\.\d+)$'
+          stream: stdout
+          linenum: 0
+      hpcg_state_stream:
+        type: str
+        regex:
+          exp: '(SUPERB)'
+          stream: stdout
+          linenum: -1
```

Comment:
this example is not very apparent since we are using `re.search` as the default if you simply remove `linenum` you will still get a match.

Since re.search will search in entire string i think metrics will still be defined. 

I would suggest you have some duplicate perhaps the content is something like

```
This is line: 1
This is line: 2
This is line: 3
This is line: 4
```

Then apply a regular expression to and specify the line number.

To showcase line number just do something like following, because the expression `This is a line` will just find the first occurence but using `linenum` you will get the specific line. The metrics `second_line` should get the string `This is a linenum: 1`

```yaml
metrics:
  second_line:
    type: str
    regex:
      exp: 'This is a line:'
      stream: stdout
      linenum: 1
  without_linenum:
    type: str
    regex:
      exp: 'This is a line:'
      stream: stdout
```

You should also test when applying regex to `stderr` in this case there is nothing written to stderr so metric should be undefined

Reply:
Can you please put this example in https://github.com/buildtesters/buildtest/tree/devel/tutorials/metrics and put a symlink in this location

Reply:
👍